core/parallel_state_processor.go

```diff
@@ -439,3 +444,20 @@ func GetDeps(txDependency [][]uint64) map[int][]int {
 
 	return deps
 }
+
+// returns true if dependencies are correct
+func VerifyDeps(deps map[int][]int) bool {
+	// number of transactions in the block
+	n := len(deps)
+
+	// Handle out-of-range and circular dependency problem
+	for tx, val := range deps {
+		for depTx := range val {
+			if depTx >= n && depTx < tx {
```

Comment:
Should this be `||` instead of `&&`?

Reply:
My bad, [done](https://github.com/maticnetwork/bor/pull/841/commits/8fe290164e80634a5c9776464f0e99aa2e86ba5e) thanks!